Improve props-II in React (#1519)
enkidevs/curriculum: javascript/react/props-ii/dealing-with-this-props-children.md

**React** provides the user with the useful class `React.Children` which provides various utilities for dealing with the opaque data structure `this.props.children`.

> `map` invokes a function on every immediate child within `children`, returning an array:

Reviewer: ```suggestion
`map` creates a new collection of children by invoking a function on every immediate child:
```

> ```jsx

Reviewer: There's no JSX code here, only JS

React.Children.map(
  children, 

>   function[(thisArg)]

Reviewer: What's the point of the `thisArg`? Not sure what this notation means.

Maybe just name the function more accurately?

```suggestion
  functionToCallOnEachChild
```

)
```

Keep in mind that if `children` is a nested object or an array it will be traversed and also that `function` will never be passed to the container objects.

> `forEach` works like `map`, yet it does not return an array:

Reviewer: ```suggestion
`forEach` works similarly to `map`, where it also iterates on each child but this time it ignores the returned value of its callback and doesn't create a new collection:
```

> ```jsx

Reviewer: This is only JS code

React.Children.forEach(
  children, 

>   function[(thisArg)]

Reviewer: ```suggestion
  functionToCallOnEachChild
```

)
```

Return the total number of components in `children`:

> ```jsx

Reviewer: JS, not JSX

React.Children.count(
  children
)
```

Return the only child in `children`:

> ```jsx

Reviewer: JS, not JSX

React.Children.only(
  children
)
```

> Manipulate collections of `children` by converting them into arrays:

Reviewer: ```suggestion
Manipulate a collection of `children` by converting it into an array:
```

> ```jsx

Reviewer: JS, not JSX
